File: API/app/Models/MedicationReminder.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class MedicationReminder extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'patient_id',
        'patient_name',
        'medication_name',
        'dose',
        'reminder_time',
        'reminder_date',
        'notes',
        'is_complete',
    ];

    protected $casts = [
        'reminder_date' => 'date',
        'reminder_time' => 'datetime:H:i',
        'is_complete' => 'boolean',
    ];

    // İlişkiler
    public function user()
    {
        return $this->belongsTo(User::class);
    }
}

File: mobile/lib/models/response_models/medication_reminder.dart
import 'package:flutter/material.dart';

class MedicationReminder {
  final String id;
  final String patientId;
  final String patientName;
  final String medicationName;
  final String dose;
  final TimeOfDay time;
  final String notes;
  bool isComplete;

  MedicationReminder({
    required this.id,
    required this.patientId,
    required this.patientName,
    required this.medicationName,
    required this.dose,
    required this.time,
    required this.notes,
    required this.isComplete,
  });
}